File: 028_map/main.cpp
#include <iostream>
#include <map>

void print_map(const std::map<char, int>& m)
{
    std::cout << "{";
    for (auto i = m.begin(); i != m.end(); ++i) {
        if (i != m.begin()) {
            std::cout << ", ";
        }
        std::cout << "'" << i->first << "': " << i->second;
    }
    std::cout << "}" << std::endl;
}

int main(void)
{
    std::cout << "initialize map" << std::endl;
    std::map<char, int> m = {{'A', 10}, {'B', 11}, {'C', 12}};
    print_map(m);
    std::cout << std::endl;

    std::cout << "get length" << std::endl;
    std::cout << "length is " << m.size() << std::endl;
    std::cout << std::endl;

    std::cout << "find" << std::endl;
    auto i = m.find('A');
    if (i != m.end()) {
        std::cout << "'A' is " << i->second << std::endl;
    }
    else {
        std::cout << "Not found 'A'" << std::endl;
    }
    std::cout << std::endl;

    std::cout << "not find" << std::endl;
    auto j = m.find('D');
    if (i != m.end()) {
        std::cout << "'D' is " << i->second << std::endl;
    }
    else {
        std::cout << "Not found 'D'" << std::endl;
    }
    std::cout << std::endl;

    std::cout << "remove" << std::endl;
    m.erase('A');
    print_map(m);
    std::cout << std::endl;

    std::cout << "range for" << std::endl;
    for (const auto& k: m) {
        std::cout << k.first << ": " << k.second << std::endl;
    }
    std::cout << std::endl;

    return 0;
}

File: 028_map/main.py
# -*- coding: utf-8 -*-

def print_map(m):
    print(m)


def main():
    print('initialize map')
    m = {'A': 10, 'B': 11, 'C': 12}
    print_map(m)
    print('')

    print('get length')
    print('length is {0}'.format(len(m)))
    print('')

    print('find')
    if 'A' in m:
        print("'A' is {0}".format(m['A']))
    else:
        print("Not found 'A'")
    print('')

    print('not find')
    if 'D' in m:
        print("'D' is {0}".format(m['D']))
    else:
        print("Not found 'D'")
    print('')

    print('remove')
    m.pop('A')
    print_map(m)
    print('')

    print('range for')
    for k, v in m.items():
        print('{0}: {1}'.format(k, v))
    print('')

if __name__ == '__main__':
    main()
